site/documentation/static/api/device-registry-v1.yaml

```diff
@@ -635,6 +635,10 @@ components:
                  "salt":
                     type: string
                     format: byte
+                 "pwd-plain":
```

Comment:
what about putting this into a separate type, e.g. `PlaintextPasswordSecret`, which doesn't have all the other properties? That would make it totally clear that a client cannot choose the hash-function etc.

Reply:
@sophokles73 I think that would spiral up in complexity (I wanted to keep it simple).

In that case we would need to have either:
1. `PlainPasswordCredential` type as well
2. or come up with a common type for "password" secrets to be able to use them both in `PasswordCredential` interchangeably. 

In order to ensure that plain password is never used in `get` operations we would need to introduce the new CredentialsSet (based on solution 1.) that would distinct if `PlainPasswordCredential` is allowed or not.

Should we go with all this or add more documentation on what are the expectation from PasswordSecret? I'm OK with both approaches in general.

Reply:
Thinking more about this, I don't think it makes much sense for management API to ever hand-over any secret specifics, like hash-functions, secrets, etc. Maybe `get` should just return `CommonSecret` data? @ctron @sophokles73 WDYT?

Reply:
Well, the use case for the *GET* operation is to retrieve the data and (potentially) edit/change it and then write it back, isn't it? For e.g. PSK I think it makes a lot of sense to get the secret in the response. You might also want to adjust/set the validity period of secrets ...

Reply:
> Should we go with all this or add more documentation on what are the expectation from PasswordSecret?

I'll leave that for you to decide. I just wanted to mention the possibility ...

Reply:
`CommonSecret` data will allow you to edit other fields, like validity period, enabled/disabled and comment. 
Maybe for pre-shared keys it makes sense, but I don't see any value in handing over password hashes other than introducing a security risk. 
These types of APIs usually don't allow you to edit passwords, just reseting them.

I need to let this brew more over the weekend :)

Reply:
@sophokles73 I ended up leaving up the same type as we had before. I think it will work better in the get/update workflow along with the other changes introduced. OpenApi doesn't define parameters dependencies yet, but it will in the future (https://swagger.io/docs/specification/describing-parameters/), which will allow us to define this better. For now we're following their recommendations and describing these dependencies.

As for the discussion on should we return secret details or not, we introduced the possibility for implementations to support both. If registry works with full secret details everything will work as it was working so far.

If registry don't want to expose secret details, it should generate an "id" for created secrets. So, get operation will return only this id and metadata for the secret. The update will merge data, so it can delete/update existing secrets or create new ones.

Let me know how this sounds to you.